### ISSUE TITLE
Tuple Covariance Crashes Intellisense Process

### ISSUE THREAD

**Issue by rbit** [microsoft/vscode-cpptools#2505]
**Describe the bug**
- OS and Version: macOS 10.13.6
- VS Code Version: 1.27.1
- C/C++ Extension Version: 0.18.1

Exercising tuple covariance will crash the intellisense process. This snippet will cause a crash:
```
#include <tuple>

void Foo(std::tuple<int>)
{}

int main(int argc, char* argv[])
{
    short val = 0;
    Foo(std::make_tuple(val));
}
```

The snippet compiles and runs without warnings:
```
clang -std=c++11 tst.cpp
```

The C/C++ output will show entries like this:
```
Checking for syntax errors: file:///Users/ray/src/gm/vscpp/tst.cpp
Failed to read response from server: 18
Checking for syntax errors: file:///Users/ray/src/gm/vscpp/tst.cpp
IntelliSense engine is not responding. Using the Tag Parser instead.
IntelliSense engine is not responding. Using the Tag Parser instead.
IntelliSense engine is not responding. Using the Tag Parser instead.
Resetting IntelliSense server: /Users/ray/src/gm/vscpp/tst.cpp
Failed to read response from server: 16
```

**Comment by sean-mcmanus** [comment]
It's not reproducing for me. What is your compilerPath and intelliSenseMode? Assuming clang, do you know what version you're using (i.e. for tuple).

**Comment by rbit** [comment]
compilerPath and intelliSenseMode are not explicitly set. For this repro, I created a new folder, and added the file tst.cpp.

System clang:
```
$ clang --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by sean-mcmanus** [comment]
Okay, thanks. I've reproed the bug. Something with the Mac C++ headers is crashing our parser (at "copy_type_with_substitution"). Looks like about 10% of Mac crashes.

UPDATE: This has been fixed with our pending 0.21.0-insiders we plan to release some time next week.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.21.0-insiders (auto installed if `"C_Cpp.updateChannel": "Insiders"` is set, or the [offline vsix](https://github.com/Microsoft/vscode-cpptools/releases/tag/0.21.0-insiders)). Let us know if you find any problems with it.
